primitives/src/targeting/eval.rs

```diff
@@ -97,8 +97,29 @@ pub enum Function {
     Max(Box<Rule>, Box<Rule>),
     Min(Box<Rule>, Box<Rule>),
     If(Box<Rule>, Box<Rule>),
+    IfNot(Box<Rule>, Box<Rule>),
+    IfElse(Box<Rule>, Box<Rule>, Box<Rule>),
     And(Box<Rule>, Box<Rule>),
+    Or(Box<Rule>, Box<Rule>),
+    Xor(Box<Rule>, Box<Rule>),
+    Not(Box<Rule>),
+    Lt(Box<Rule>, Box<Rule>),
+    Lte(Box<Rule>, Box<Rule>),
+    Gt(Box<Rule>, Box<Rule>),
+    Gte(Box<Rule>, Box<Rule>),
+    Equals(Box<Rule>, Box<Rule>),
```

Comment:
Keep in mind that this will become `equals: {...}` and not the DSL `eq`.
This is due to `#[serde(rename_all = "camelCase")]`, by default it uses the enum variant as a tag (see https://serde.rs/enum-representations.html).
This also includes `NotEquals`, `NotIn`.
You can use (see https://serde.rs/attributes.html):
```rust
#[derive(Serialize, Deserialize)]
#[serde(rename = "e")]  // <-- this is also a container attribute
enum E {
    #[serde(rename = "a")]  // <-- this is a variant attribute
    A(String),
}
```

however it's maybe better to Document (use doc comments to explain what is `eq`, `nin`, etc.) which type is what and keep the names the same as the DSL names.